src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

```diff
@@ -310,6 +314,23 @@ private async void FileList_SelectionChanged(object sender, SelectionChangedEven
 			{
 				await QuickLookHelpers.ToggleQuickLook(ParentShellPageInstance, true);
 			}
+
+
+			// If the selection is not all items, uncheck the select all checkbox and vice versa
+			SelectAllItemsCheckbox.IsChecked = SelectedItems.Count == FileList.Items.Count;
```

Comment:
Creating a property and binding the checkbox to it might be a little more future proof.

Reply:
Good point, updated that now.